packages/web-app-files/src/components/SideBar/Shares/Collaborators/CollaboratorsEditOptions.vue

```diff
@@ -1,32 +1,50 @@
 <template>
   <div>
-    <translate tag="label" for="files-collaborators-role-button" class="oc-label">Role</translate>
-    <oc-select
-      v-model="selectedRole"
-      input-id="files-collaborators-role-button"
-      class="files-collaborators-role-button-wrapper"
-      :options="roles"
-      :clearable="false"
-      label="label"
+    <hr />
+    <oc-button
+      id="files-collaborators-role-button"
+      appearance="raw"
+      justify-content="left"
+      gap-size="xsmall"
     >
-      <template #option="option">
-        <role-item :role="option" />
+      <translate v-if="isAdvancedRoleSelected" key="advanced-permissions-select"
+        >Invite with custom permissions</translate
+      >
+      <translate v-else key="role-select" :translate-params="{ name: selectedRole.inlineLabel }"
+        >Invite as %{ name }</translate
+      >
+      <oc-icon name="expand_more" />
+    </oc-button>
+    <oc-drop toggle="#files-collaborators-role-button" mode="click" close-on-click>
+      <template #special>
+        <oc-list class="files-recipient-role-drop-list" :aria-label="rolesListAriaLabel">
+          <li v-for="role in roles" :key="role.name">
+            <oc-button
+              appearance="raw"
+              justify-content="space-between"
+              class="files-recipient-role-drop-btn oc-py-xs oc-px-s"
+              :class="{ selected: isSelectedRole(role) }"
+              @click="selectRole(role)"
+            >
+              <role-item :role="role" />
+              <oc-icon v-if="isSelectedRole(role)" name="check" />
+            </oc-button>
+          </li>
+        </oc-list>
       </template>
-      <template #no-options v-translate>
-        No matching role found
-      </template>
-    </oc-select>
+    </oc-drop>
     <template v-if="$_ocCollaborators_hasAdditionalPermissions">
-      <label v-if="selectedRole.name !== 'advancedRole'" class="oc-label oc-mt-s">
+      <label v-if="!isAdvancedRoleSelected" class="oc-label oc-mt-s">
         <translate>Additional permissions</translate>
       </label>
       <additional-permissions
         :available-permissions="selectedRole.additionalPermissions"
         :collaborators-permissions="collaboratorsPermissions"
-        :class="{ 'oc-mt-s': selectedRole.name === 'advancedRole' }"
+        :class="{ 'oc-mt-s': isAdvancedRoleSelected }"
         @permissionChecked="checkAdditionalPermissions"
       />
     </template>
+    <hr />
```

Comment:
Probably only should be shown if the datepicker is present to avoid having two `<hr>`s with no content in between (line 37 in `EditCollaborator.vue)